icu4c/source/test/intltest/numfmtst.cpp

```diff
@@ -10063,4 +10064,31 @@ void NumberFormatTest::Test21232_ParseTimeout() {
     // Should not hang
 }
 
+void NumberFormatTest::Test10997_FormatCurrency() {
+    IcuTestErrorCode status(*this, "Test10997_FormatCurrency");
+
+    UErrorCode error = U_ZERO_ERROR;
+    NumberFormat* fmt = NumberFormat::createCurrencyInstance(Locale::getUS(), error);
+    fmt->setMinimumFractionDigits(4);
+    fmt->setMaximumFractionDigits(4);
+
+    FieldPosition fp;
+
+    UChar usd[] = { 'U', 'S', 'D', 0 };
```

Comment:
Better: `const char16_t* USD = u"USD";` or just put `u"USD"` inline below

Reply:
No problem. I copied that style from another test.